seqr/views/utils/dataset_utils.py

```diff
@@ -261,28 +263,41 @@ def match_and_update_search_samples(
 
 
 def _match_and_update_rna_samples(
-    projects, user, sample_ids, data_source, sample_id_to_individual_id_mapping, raise_unmatched_error_template,
+    projects, user, sample_project_tuples, data_source, sample_id_to_individual_id_mapping, raise_unmatched_error_template,
 ):
-    samples = Sample.objects.select_related('individual').filter(
+    sample_project_lookup = defaultdict(set)
+    for sample_id, project_name in sample_project_tuples:
+        sample_project_lookup[sample_id].add(project_name)
+    no_project_key = next(iter(list(sample_project_tuples)), ('', None))[1] is None
```

Comment:
After updating Outlier data to take in project names, we can eliminate `no_project_key`.

Reply:
we should not add clunky code we plan to immediately remove. Just add outlier as part of this PR, or do outlier branched off this PR so it can be reviewed separately but then merge it here before this goes to dev

Reply:
I have created a [PR](https://github.com/broadinstitute/seqr/pull/3287) for Outliers.

Reply:
okay I will review this part of the code again once we have merged that to this branch